src/sqlparser/src/ast/statement.rs

```diff
@@ -847,19 +842,23 @@ pub struct CsvInfo {
     pub has_header: bool,
 }
 
+pub fn get_delimiter(chars: &str) -> Result<u8, ParserError> {
```

Comment:
I guess we can directly call `unescape_c_style` here to match the behavior of how we handle string literal, which is currently done in the binder.

Can we directly `unescape` before returning `CstyleEscapesString` in the tokenizer? It seems confusing that the inner string is not escaped at all. cc @xiangjinwu @ZENOTME

Reply:
> Can we directly `unescape` before returning `CstyleEscapesString` in the tokenizer?

I also thought about this when reviewing the sqlparser PR, then realized we would need to escape it again when `unparse` or `Display`.

Reply:
Or we may also store the original form? 😂

Reply:
You meant store both? The current implementation stores the original form only, resulting in binder and this PR required to `unescape` before consumption.

I feel like it may be better to store `Value::CstyleEscapesString(CstyleEscapesString)` rather than `CstyleEscapesString(String)`, and the new type `Display`s as `E'raw'` and has a `.real_value(&self) -> String` to unescape.

Reply:
Yeah, exactly!